addons/point_of_sale/point_of_sale.py

```diff
@@ -39,6 +39,45 @@ def _get_currency(self, cr, uid, ids, fieldnames, args, context=None):
             result[pos_config.id] = currency_id
         return result
 
+    def _get_current_session(self, cr, uid, ids, fieldnames, args, context=None):
+        result = dict()
+
+        for record in self.browse(cr, uid, ids, context=context):
+            session_id = record.session_ids.filtered(lambda r: r.user_id.id == uid and not r.state == 'closed')
+            result[record.id] = {
+                'current_session_id': session_id,
+                'current_session_state': session_id.state,
```

Comment:
I doubt this will work if there is no session matching the filter. In this case session_id will be an empty  recordset, so I don't think accessing its state will work, could you please test it ?